lib/tasks/data/truncate_data.rake

```diff
@@ -1,81 +1,46 @@
+#
+# This will remove data older than 2 years (24 months).
 namespace :ofn do
   namespace :data do
     desc 'Truncate data'
-    task truncate: :environment do
-      guard_and_warn
-
-      sql_delete_from "
-        spree_inventory_units #{where_order_id_in_orders_to_delete}"
-      sql_delete_from "
-        spree_inventory_units
-        where shipment_id in (select id from spree_shipments #{where_order_id_in_orders_to_delete})"
-
-      truncate_adjustments
-
-      sql_delete_from "spree_line_items #{where_order_id_in_orders_to_delete}"
-      sql_delete_from "spree_payments #{where_order_id_in_orders_to_delete}"
-      sql_delete_from "spree_shipments #{where_order_id_in_orders_to_delete}"
-      Spree::ReturnAuthorization.delete_all
-
-      truncate_order_cycle_data
-
-      sql_delete_from "proxy_orders #{where_oc_id_in_ocs_to_delete}"
-
-      sql_delete_from "spree_orders #{where_oc_id_in_ocs_to_delete}"
-      sql_delete_from "order_cycle_schedules #{where_oc_id_in_ocs_to_delete}"
-      sql_delete_from "order_cycles #{where_ocs_to_delete}"
-
-      Spree::TokenizedPermission.where("created_at < '#{date}'").delete_all
-      Spree::StateChange.delete_all
-      Spree::LogEntry.delete_all
-      sql_delete_from "sessions"
-    end
+    task :truncate, [:months_to_keep] => :environment do |_task, args|
+      warn_with_confirmation
 
-    def sql_delete_from(sql)
-      ActiveRecord::Base.connection.execute("delete from #{sql}")
+      months_to_keep = args.months_to_keep.to_i
+      TruncateData.new(months_to_keep).call
     end
 
-    private
-
-    def date
-      3.months.ago
-    end
-
-    def where_ocs_to_delete
-      "where orders_close_at < '#{date}'"
-    end
-
-    def where_oc_id_in_ocs_to_delete
-      "where order_cycle_id in (select id from order_cycles #{where_ocs_to_delete} )"
-    end
-
-    def where_order_id_in_orders_to_delete
-      "where order_id in (select id from spree_orders #{where_oc_id_in_ocs_to_delete})"
-    end
-
-    def truncate_adjustments
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Order'
-        and source_id in (select id from spree_orders #{where_oc_id_in_ocs_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Shipment'
-        and source_id in (select id from spree_shipments #{where_order_id_in_orders_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::Payment'
-        and source_id in (select id from spree_payments #{where_order_id_in_orders_to_delete})"
-      sql_delete_from "spree_adjustments where source_type = 'Spree::LineItem'
-        and source_id in (select id from spree_line_items #{where_order_id_in_orders_to_delete})"
-    end
+    def warn_with_confirmation
+      message = <<-MSG.strip_heredoc
+      \n
+      <% highlighted_message = "This will permanently change DB contents. Please, make a backup first." %>
+      <%= color(highlighted_message, :blink, :on_red) %>
```

Comment:
This now looks like

![Screenshot from 2020-03-16 18-25-14](https://user-images.githubusercontent.com/762088/76784289-a5ff4b80-67b3-11ea-9920-c46664f06a78.png)

Reply:
:-)